### PR TITLE
Code snippet question

### PR DESCRIPTION
**Pull request by Deepankar-promact**
Created models for the code snippet question.

### CODE REVIEW

File: Trappist/src/Promact.Trappist.DomainModel/Data/SeedLanguage.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    /// Polpulates pre-required table
+    /// </summary>
+    public class SeedDatabase
+    {
+        /// <summary>
+        /// Seed data into the table
+        /// </summary>
+        /// <param name="context">DbContext</param>
+        public static void Seed(TrappistDbContext context) {
+            context.Database.EnsureCreated();
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Data/SeedLanguage.cs]: What is the significance of this line over here?

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodeSnippetQuestion.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodeSnippetQuestion.cs]: Please remove unused using statements.

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodingLanguage.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Promact.Trappist.DomainModel.Models.Question
+{
+    public enum language: int
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodingLanguage.cs]: enum should be in the separate file and enum name is not proper

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionLanguageMapping.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        public int QuestionId { get; set; }
+
+        [Required]
+        public int LanguageId { get; set; }
+
+        [ForeignKey("QuestionId")]
+        public CodeSnippetQuestion CodeSnippetQuestion { get; set; }
+
+        [ForeignKey("LanguageId")]
+        public CodingLanguage CodeLanguage { get; set; }
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionLanguageMapping.cs]: virtual keyword is missing.

File: Trappist/src/Promact.Trappist.Web/Migrations/20170309094707_migration-csq-4.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Promact.Trappist.Web.Migrations
+{
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/Migrations/20170309094707_migration-csq-4.cs]: why there is empty migration file?

File: Trappist/src/Promact.Trappist.DomainModel/Data/SeedLanguage.cs (hunk 1 above)
Comment (Deepankar-promact) [inline Trappist/src/Promact.Trappist.DomainModel/Data/SeedLanguage.cs]: Database.EnsureCreated function create the database if its not created.

File: Trappist/src/Promact.Trappist.Web/Migrations/20170310033212_migration-csq-1.cs [hunk 6]
```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore.Migrations;
+using Microsoft.EntityFrameworkCore.Metadata;
+
+namespace Promact.Trappist.Web.Migrations
+{
+    public partial class migrationcsq1 : Migration
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Web/Migrations/20170310033212_migration-csq-1.cs]: MINOR Rename class 'migrationcsq1' to match camel case naming rules, consider using 'Migrationcsq1'. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS101)

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClass/ProgramingLanguageEnum.cs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Promact.Trappist.DomainModel.ApplicationClass
+{
+    public enum ProgramingLanguageEnum
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClass/ProgramingLanguageEnum.cs]: MINOR Rename this enumeration to remove the 'Enum' suffix. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS2344)

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClass/SeedDatabase.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+using System;
+using System.Linq;
+
+namespace Promact.Trappist.DomainModel.Data
+{
+
+    /// <summary>
+    /// Polpulates pre-required table
+    /// </summary>
+    public class SeedDatabase
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClass/SeedDatabase.cs]: MAJOR Add a 'protected' constructor or the 'static' keyword to the class declaration. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1118)

**Comment by chintans** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TrappistDbContext.cs#L24](https://github.com/Promact/trappist/blob/bd239e9dcc48cb49a6398ca4d9a91dcf8ad2ecec/Trappist/src/Promact.Trappist.DomainModel/DbContext/TrappistDbContext.cs#L24): Remove this commented out code. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS125)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**